cmd/diff.go

```diff
@@ -69,33 +69,58 @@ func checkFilenameFlag(_ []string) error {
 	return errors.New("Please include --types=file with the --filename flag")
 }
 
+func processImage(imageName string, imageMap map[string]*pkgutil.Image, wg *sync.WaitGroup, errChan chan<- error) {
+	defer wg.Done()
+	image, err := getImageForName(imageName)
+	if image.Image == nil {
```

Comment:
Ok this logic isn't great so I'm gonna rework it a bit. Digging into the go-containerregistry code, if we get an error returned back from any of the three sources (tar/daemon/remote), the image returned will always be `nil`. So it's equivalent to just change this back to a vanilla `if err != nil` check.

The reason I was checking for `image.Image == nil` is to avoid putting the image into the returned map so we didn't attempt to clean up an empty path, but that logic is already happening in `pkgutil.CleanupImage` so this is safe. I moved the deferred calls up before we return the errors, and we're good.

TLDR answers for your questions: 1) yes, and 2) not possible